kolibri/plugins/coach_tools/assets/src/vue/reports/index.vue

```diff
@@ -155,38 +157,15 @@
         });
       },
       allViewLink() {
-        return this.genLink({ all_or_recent: Constants.AllOrRecent.ALL });
-      },
-    },
-    methods: {
-      /* Generates a REPORTS link relative to the current page, with only newParams changed. */
-      genLink(newParams) {
-        const currentParams = {
-          channel_id: this.pageState.channel_id,
-          content_scope: this.pageState.content_scope,
-          content_scope_id: this.pageState.content_scope_id,
-          user_scope: this.pageState.user_scope,
-          user_scope_id: this.pageState.user_scope_id,
-          all_or_recent: this.pageState.all_or_recent,
-          view_by_content_or_learners: this.pageState.view_by_content_or_learners,
-          sort_column: this.pageState.sort_column,
-          sort_order: this.pageState.sort_order,
-        };
-        const vlink = {
-          name: Constants.PageNames.REPORTS,
-          params: {},
-        };
-        Object.assign(vlink.params, currentParams, newParams);
-        return vlink;
+        return genLink(this.pageState, { all_or_recent: Constants.AllOrRecent.ALL });
       },
-      switchView(isContent) {
-        const newView = isContent ? Constants.ViewBy.CONTENT : Constants.ViewBy.LEARNERS;
-        this.$router.go(this.genLink({ view_by_content_or_learners: newView }));
+      viewByLink() {
+        const view = this.isViewByContent ? Constants.ViewBy.LEARNERS : Constants.ViewBy.CONTENT;
```

Comment:
no - target of the link is the opposite of the current view. I'll add a clarifying comment